modules/tensor_mechanics/src/bcs/Torque.C

```diff
@@ -0,0 +1,131 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "Torque.h"
+#include "Function.h"
+
+registerMooseObject("TensorMechanicsApp", Torque);
+registerMooseObject("TensorMechanicsApp", ADTorque);
+
+template <bool is_ad>
+InputParameters
+TorqueTempl<is_ad>::validParams()
+{
+  InputParameters params = TorqueParent<is_ad>::validParams();
+  params.addClassDescription(
+      "Apply a moment as tractions distributed over a surface around a "
+      "pivot point. This should operate on the displaced mesh for large deformations.");
+  MooseEnum component("x=0 y=1 z=2");
+  params.addRequiredParam<Point>("origin", "Pivot point for the twisting moment");
```

Comment:
Do we need to allow for the origin to move for finite deformations? I guess there are relevant cases where it doesn't move, but it seems like that might be important for some cases.

Reply:
The origin should be the `Point` equivalent of a Postprocessor (using the new Reporter system).  But I would wait implementing it that way until users can specify value literals in place of reporter names.